### ISSUE TITLE
Implement Patch Test page

### ISSUE THREAD

**Issue by pingdynasty** [pingdynasty/OwlServer#75]
We have the ability to generate a JS version of the patch (`patch.js`) at compilation time.
If the `patch.js` file is available and the browser supports Web audio, the user should be able run it in the patch details page.
This could be integrated into the design as tabbed content along with the soundcloud sample and/or the parameter view.
In Patch Test, the user should have control over the 5 parameters and push button.
There should be a stop/start (or play/pause) toggle and a drop-down to select audio input from one of:
* none
* selection of wavs (acoustic/electric guitar, drums, synth, swept sine, noise)
* mic

Blocked by https://github.com/pingdynasty/OwlProgram/issues/15

**Comment by pingdynasty** [comment]
An implied additional task here is to redesign (parts of) the patch details page. Are you happy to take that on, @samuele-artuso ?

**Comment by samuele-artuso** [comment]
> Are you happy to take that on, samuele-artuso ?

Yes, I am!

**Comment by pingdynasty** [comment]
@tamagazzi Tom: can you come up with some good samples to play through the patch?
I think we must check the number of inputs the patch has, and only connect stereo if it has two in.
This way all the samples must be stereo (or dual mono) but will work with both stereo and mono patches.

**Comment by samuele-artuso** [comment]
Could you guys get me the samples?

**Comment by samuele-artuso** [comment]
Note to self:
* implement tabbed view (info/test);
* ability to stop patch;
* when a JS build is not available, ask the user if they want to compile the patch.

**Comment by samuele-artuso** [comment]
On staging.

**Comment by pingdynasty** [comment]
Looks good, ready to deploy?

I think there are issues with mono patches which we can sort out later.
Can you connect/disconnect the channels separately?

Also I get a warning in the console: getUserMedia() is deprecated on insecure origins. This is for loading the input samples, right?

**Comment by samuele-artuso** [comment]
Pushed live.

> Can you connect/disconnect the channels separately?

I can see (actually: hear) the issue with mono patches. The patch will play on the left channel and the sample on the right one. Is it the same issue you're experiencing?

> Also I get a warning in the console: getUserMedia() is deprecated on insecure origins.

I think this has got to do with the fact our website doesn't run on HTTPS. More info [here](https://sites.google.com/a/chromium.org/dev/Home/chromium-security/deprecating-powerful-features-on-insecure-origins).

There's another issue in that the `getUserMedia()` API call is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getUserMedia) and will be removed altogether. I think I can fix this, but I'm afraid we'll keep on getting that warning.

**Comment by samuele-artuso** [comment]
> This is for loading the input samples, right?

It's for asking users to access their mic.

**Comment by pingdynasty** [comment]
done.
